quantum/impl/quantum_read_write_spinlock_impl.h

```diff
@@ -18,6 +18,43 @@
 //##############################################################################################
 //#################################### IMPLEMENTATIONS #########################################
 //##############################################################################################
+/*
+The atomic 32-bit counter is divided into two halves, the high 16 bits representing the number of pending
+writer upgrades and the low 16 bit representing the current number of reader owners or -1
```

Comment:
"representing the current number of reader owners or -1" why is it -1 but not 0? What does 0 mean currently?

Reply:
-1 means write locked. 0 means unlocked and >0 means read locked. I will update the comment better.

Reply:
Ok I improved description.